src/main/java/com/squareup/kotlinpoet/TypeAliasSpec.kt

```diff
@@ -97,7 +95,18 @@ class TypeAliasSpec private constructor(builder: TypeAliasSpec.Builder) {
       kdoc.add(block)
     }
 
-    fun build() = TypeAliasSpec(this)
+    fun build(): TypeAliasSpec {
+      modifiers.forEach {
+        require(it in ALLOWABLE_MODIFIERS) {
+          "unexpected typealias modifier $it"
+        }
+      }
+      return TypeAliasSpec(this)
+    }
+
+    private companion object {
+      private val ALLOWABLE_MODIFIERS = setOf(PUBLIC, INTERNAL, PRIVATE, ACTUAL)
```

Comment:
nit: shouldn't it be ALLOWED? non-native speaker here, ignore me if that's the correct wording

Reply:
I think either work :)